### ISSUE TITLE
Bad interaction of %ignore with %rename for template classes methods

### ISSUE THREAD

**Issue by vadz** [swig/swig#85]
This is a strange bug which is probably best shown with an example:

```
$ cat -n ut.i
     1  %module ut
     2
     3  %ignore *::Foo;
     4  %rename("%s") T::Foo;
     5  //%rename("%s") T<int>::Foo;
     6
     7  %inline %{
     8    template <typename V> struct T {
     9        V Foo();
    10        V Bar();
    11    };
    12
    13    struct S {
    14      void Foo();
    15      void Bar();
    16    };
    17  %}
    18  %template(IntT) T<int>;
$ swig -csharp -c++ -o wrap_ut_csharp.cpp ut.i
$ fgrep Foo *.cs || echo "Foo not exported"
Foo not exported
```

Uncommenting line 5 makes it work as expected, i.e. `T::Foo()` is exported. But both line 5 _and_ line 4 are needed, if either one is omitted, `Foo()` is ignored.

Also notice that this is related to `%ignore`, without it, i.e. if we use `%rename(Baz)` on `Foo()`, line 4 on its own works just fine.

This was first reported in [this thread](http://thread.gmane.org/gmane.comp.programming.swig.devel/22633)

**Comment by ojwb** [comment]
Still present in c51ce4995c07ada09827aaa9c062f3910dec231b (before 3.0.10).

**Comment by ojwb** [comment]
Seems to have been been fixed in the meantime - with SWIG 4.0.2 the `fgrep` gives:

```
IntT.cs:  public int Foo() {
IntT.cs:    int ret = utPINVOKE.IntT_Foo(swigCPtr);
utPINVOKE.cs:  [global::System.Runtime.InteropServices.DllImport("ut", EntryPoint="CSharp_IntT_Foo")]
utPINVOKE.cs:  public static extern int IntT_Foo(global::System.Runtime.InteropServices.HandleRef jarg1);
```
